### PR TITLE
feat(3068): flatten pipeline template

### PR DESCRIPTION
**Pull request by klu909**
## Context

flatten pipeline template by eliminating shared settings 
## Objective
remove shared settings from merging logic and flatten shared into jobs

## References
https://github.com/screwdriver-cd/screwdriver/issues/3068

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: lib/phase/merge.js [hunk 1]
```diff
@@ -161,36 +161,28 @@ function mergePipelineLevelConfig(parsedDoc, pipelineTemplate) {
...
-    if (mergedImage) {
-        newPipeline.shared.image = mergedImage;
+    if (yamlSharedConfig.image) {
+        newPipeline.shared.image = yamlSharedConfig.image;
     }
 
     newPipeline.templateVersionId = pipelineTemplate.id;
 
+    console.log(newPipeline);
+
```
Comment (tkyi) [inline lib/phase/merge.js]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "screwdriver-config-parser",
-  "version": "8.0.0",
+  "version": "10.0.0",
```
Comment (VonnyJap) [inline package.json:3]:
```suggestion
  "version": "9.0.0",
```

File: index.js [hunk 3]
```diff
@@ -308,4 +310,30 @@ module.exports = function configParser({
...
+        abortEarly: false
+    });
+    const pipelineTemplateConfig = pipelineTemplate.config;
+
+    // flatten pipeline template shared setting into jobs
+    pipelineTemplateConfig.jobs = flattenSharedIntoJobs(pipelineTemplateConfig.shared, pipelineTemplateConfig.jobs);
+    delete pipelineTemplateConfig.shared;
+
+    return pipelineTemplate;
+}
```
Comment (VonnyJap) [inline index.js:334]: where would this function be used? does this function need to be incorporated as well into `screwdriver-models` package?

File: package.json (hunk 2 above)
Comment (klu909) [reply package.json:3]: we are already at major version 9, it just didn't get changed last time. https://www.npmjs.com/package/screwdriver-config-parser

File: index.js (hunk 3 above)
Comment (klu909) [reply index.js:334]: validator will call this, then screwdriver api calls validator https://github.com/screwdriver-cd/screwdriver/blob/11813f8ac54a3b3bed4e4298ce7ba754c7b7d992/plugins/pipelines/templates/create.js#L22

**Review by VonnyJap — APPROVED** [review]

**Comment by VonnyJap** [comment]
@klu909 - PR is approved, please help to propagate the change to other affected modules. 
And another thing, do we need to remove shared from [here](https://github.com/screwdriver-cd/data-schema/blob/54a8ab5985dbe76b456f07e1dc2e75dabbfc1dbb/config/pipelineTemplate.js#L13)?

**Comment by klu909** [comment]
> here

don't think we can, initial user input will have `shared`, we just don't store them

**Pull request merged**